Connection.php

```diff
@@ -436,18 +436,25 @@ protected function _run(HoaSocket\Node $node)
                         }
                     }
 
-                    $this->close(self::CLOSE_NORMAL);
+                    
                     $this->getListener()->fire(
                         'close',
                         new Event\Bucket([
                             'code'   => $code,
                             'reason' => $reason
                         ])
                     );
-
+                    $this->close(self::CLOSE_NORMAL);
                     break;
 
                 default:
+                    $this->getListener()->fire(
+                        'close',
+                        new Event\Bucket([
+                            'code'   =>self::CLOSE_PROTOCOL_ERROR,
```

Comment:
@Hywan Hi, I'm interested on how you do that (the git diff), can you tell me please ?

By the way, sorry to spam this thread, delete this comment if you want.

Mickaël